--- a/build/linux/Makefile	2018-11-12 03:26:56.000000000 +0800
+++ b/build/linux/Makefile	2018-11-12 03:27:02.000000000 +0800
@@ -163,19 +163,19 @@

 # AArch64 NEON optims
 ifeq ($(SYS_ARCH),AARCH64)
-ASMSRC += common/aarch64/bitstream-a.S \
-          common/aarch64/cabac-a.S     \
-          common/aarch64/dct-a.S     \
-          common/aarch64/deblock-a.S \
-          common/aarch64/mc-a.S      \
-          common/aarch64/pixel-a.S   \
-          common/aarch64/predict-a.S \
-          common/aarch64/quant-a.S
-SRCS   += common/aarch64/asm-offsets.c \
-          common/aarch64/mc-c.c        \
-          common/aarch64/predict-c.c
-OBJASM  = $(ASMSRC:%.S=%.o)
-OBJCHK += tools/checkasm-aarch64.o
+#ASMSRC += common/aarch64/bitstream-a.S \
+#          common/aarch64/cabac-a.S     \
+#          common/aarch64/dct-a.S     \
+#          common/aarch64/deblock-a.S \
+#          common/aarch64/mc-a.S      \
+#          common/aarch64/pixel-a.S   \
+#          common/aarch64/predict-a.S \
+#          common/aarch64/quant-a.S
+#SRCS   += common/aarch64/asm-offsets.c \
+#          common/aarch64/mc-c.c        \
+#          common/aarch64/predict-c.c
+#OBJASM  = $(ASMSRC:%.S=%.o)
+#OBJCHK += tools/checkasm-aarch64.o
 endif
 
 # MSA optims
